app/services/candidate_interface/find_changed_apply_again_applications.rb

```diff
@@ -0,0 +1,48 @@
+module CandidateInterface
+  class FindChangedApplyAgainApplications
+    def all_forms
+      apply_again_forms
+    end
+
+    def changed_forms
+      apply_again_forms.find_each.lazy.select do |application_form|
+        changed?(application_form)
+      end
+    end
+
+    def all_candidate_count
+      all_forms.map(&:candidate_id).uniq.count
+    end
+
+    def changed_candidate_count
+      changed_forms.map(&:candidate_id).uniq.count
+    end
+
+  private
+
+    def apply_again_forms
+      ApplicationForm
+        .where(
+          phase: 'apply_2',
+          recruitment_cycle_year: RecruitmentCycle.current_year,
+        )
+        .includes(:application_qualifications, previous_application_form: [:application_qualifications])
+    end
+
+    def changed?(application_form)
+      significant_values(application_form) !=
+        significant_values(application_form.previous_application_form)
+    end
+
+    def significant_values(application_form)
+      application_form.as_json(
+        only: %i[becoming_a_teacher subject_knowledge],
+        include: {
+          application_qualifications: {
```

Comment:
Good question. I think it's OK because `Hash#==` ignores order (even though hashes are ordered in Ruby). If they were arrays we'd be in trouble.

```
pry(main)> { b: 'B', a: 'A', c: { d: 'D', e: 'E' } } == { a: 'A', c: { e: 'E', d: 'D' }, b: 'B' }
=> true
```